ntp-daemon/src/peer.rs

```diff
@@ -92,36 +89,15 @@ where
             .reset(self.last_poll_sent + poll_interval);
     }
 
-    fn fitness(
-        &self,
-        ntp_instant: NtpInstant,
-        system_poll: PollInterval,
-    ) -> Result<(), AcceptSynchronizationError> {
-        self.peer.accept_synchronization(
-            ntp_instant,
-            self.config.frequency_tolerance,
-            self.config.distance_threshold,
-            system_poll.as_duration(),
-        )
-    }
-
     async fn handle_poll(&mut self, poll_wait: &mut Pin<&mut Sleep>) {
-        let ntp_instant = NtpInstant::now();
-
         let system_snapshot = *self.channels.system_snapshots.read().await;
         let packet = self.peer.generate_poll_message(system_snapshot);
 
         // Sent a poll, so update waiting to match deadline of next
         self.last_poll_sent = Instant::now();
         self.update_poll_wait(poll_wait, system_snapshot);
 
-        let accept = self.fitness(ntp_instant, system_snapshot.poll_interval);
-
-        if let Err(accept_error) = accept {
-            info!(?accept_error, "peer is not fit for use in synchronization");
-            let msg = MsgForSystem::PeerNotFit(self.index);
-            self.channels.msg_for_system_sender.send(msg).await.ok();
-        }
+        // NOTE: fitness check is not performed here, but by System
```

Comment:
This now removes the entire step of sending here, reintroducing the bug of us considering unreachable peers.

Reply:
That would be best. I would consider just having two snapshot-sending messages, one where we have an actual new measurement, and one where we do not. And then even on the new measurement one system should probably only do clock selection and such if the peer is acceptable after this new measurement.